ui/page/dcrdex/dcrdex_page.go

```diff
@@ -21,66 +25,97 @@ type DEXPage struct {
+	dexCore Core
+	inited  bool // TODO: Set value
 
-	openTradeMainPage *cryptomaterial.Clickable
-	inited            bool // TODO: Set value
+	generalSettingsBtn cryptomaterial.Button
+	openTradeMainPage  *cryptomaterial.Clickable
 }
 
-func NewDEXPage(l *load.Load) *DEXPage {
-	dp := &DEXPage{
-		Load:              l,
-		MasterPage:        app.NewMasterPage(DCRDEXID),
-		openTradeMainPage: l.Theme.NewClickable(false),
+func NewDEXPage(l *load.Load, dexCore Core) *DEXPage {
+	pg := &DEXPage{
+		Load:               l,
+		MasterPage:         app.NewMasterPage(DCRDEXPageID),
+		openTradeMainPage:  l.Theme.NewClickable(false),
+		dexCore:            dexCore,
+		generalSettingsBtn: l.Theme.Button(values.StringF(values.StrEnableAPI, values.String(values.StrExchange))),
 	}
-	return dp
+
+	return pg
 }
 
 // ID is a unique string that identifies the page and may be used to
 // differentiate this page from other pages.
 // Part of the load.Page interface.
 func (pg *DEXPage) ID() string {
-	return DCRDEXID
+	return DCRDEXPageID
 }
 
 // OnNavigatedTo is called when the page is about to be displayed and may be
 // used to initialize page features that are only relevant when the page is
 // displayed.
 // Part of the load.Page interface.
 func (pg *DEXPage) OnNavigatedTo() {
-	pg.ctx, pg.ctxCancel = context.WithCancel(context.TODO())
-
 	if pg.CurrentPage() == nil {
 		// TODO: Handle pg.inited
-		pg.Display(NewDEXOnboarding(pg.Load))
+		pg.Display(NewDEXOnboarding(pg.Load, pg.dexCore))
+	} else {
+		pg.CurrentPage().OnNavigatedTo()
 	}
-
-	pg.CurrentPage().OnNavigatedTo()
 }
 
 // Layout draws the page UI components into the provided layout context to be
 // eventually drawn on screen.
 // Part of the load.Page interface.
 func (pg *DEXPage) Layout(gtx C) D {
-	return layout.Stack{}.Layout(gtx,
-		layout.Expanded(func(gtx C) D {
-			return cryptomaterial.LinearLayout{
-				Width:       cryptomaterial.MatchParent,
-				Height:      cryptomaterial.MatchParent,
-				Orientation: layout.Vertical,
-			}.Layout(gtx,
-				layout.Flexed(1, pg.CurrentPage().Layout),
-			)
-		}),
-	)
+	hasMultipleWallets := pg.isMultipleAssetTypeWalletAvailable()
+	privacyModeOff := pg.WL.AssetsManager.IsHTTPAPIPrivacyModeOff(utils.ExchangeHTTPAPI)
+	var msg string
+	var actionBtn *cryptomaterial.Button
+	if !hasMultipleWallets {
+		msg = values.String(values.StrMultipleAssetTypeRequiredToTradeDEX)
+	} else if !privacyModeOff {
+		actionBtn = &pg.generalSettingsBtn
+		msg = values.StringF(values.StrNotAllowed, values.String(values.StrExchange))
+	} else if dexCore, _ := pg.dexCore.(*dexcore.DEXCore); dexCore == nil {
+		msg = values.String(values.StrDEXInitErrorMsg)
+	}
+
+	if msg != "" {
+		return components.DisablePageWithOverlay(pg.Load, nil, gtx, msg, actionBtn)
+	}
+
+	return cryptomaterial.LinearLayout{
+		Width:       cryptomaterial.MatchParent,
+		Height:      cryptomaterial.MatchParent,
+		Orientation: layout.Vertical,
+	}.Layout2(gtx, pg.CurrentPage().Layout)
```

Comment:
Please, what does `Layout2` do and how is it different from `Layout`?

Also, why is pg.CurrentPage() being wrapped in a LinearLayout? Any `pg.CurrentPage()` that needs its widgets to be wrapped in this way should do the wrapping themselves, this should probably just be `pg.CurrentPage().Layout(gtx)`, unless I'm missing something.

Reply:
> this should probably just be pg.CurrentPage().Layout(gtx)

I thought about this but wanted a way to ensure all child elements are displayed vertically. But I think we can return `pg.CurrentPage().Layout(gtx)` as you've suggested, all pages should proly handle orientation themselves.